hat/assets/js/apps/Iaso/domains/mappings/details.js

```diff
@@ -58,233 +54,182 @@ const fieldTypeOptions = formatMessage => [
 
-        const onQuestionSelected = node => {
-            setCurrentQuestion(node);
-            redirectToReplace(baseUrls.mappingDetail, {
-                mappingVersionId: currentMappingVersion.id,
-                questionName: node.name,
-            });
-        };
+    const currentFormVersion = currentMappingVersionQuery?.data?.formVersion;
 
-        const onConfirmedQuestionMapping = questionMapping => {
-            applyPartialUpdate(
-                currentMappingVersion.id,
-                currentQuestion.name,
-                questionMapping,
-            );
-        };
+    const fetching = currentMappingVersionQuery.isFetching;
 
-        const onUnmapQuestionMapping = () => {
-            applyPartialUpdate(currentMappingVersion.id, currentQuestion.name, {
-                action: 'unmap',
-            });
-        };
+    const indexedQuestions = currentFormVersion
+        ? Descriptor.indexQuestions(currentFormVersion.descriptor)
+        : {};
 
-        const onNeverMapQuestionMapping = () => {
-            applyPartialUpdate(currentMappingVersion.id, currentQuestion.name, {
-                type: 'neverMapped',
-            });
-        };
-        const isDataElementMappable =
-            currentMappingVersion &&
-            currentMappingVersion.mapping.mapping_type !== 'DERIVED';
-        const indexedQuestions =
-            currentFormVersion &&
-            Descriptor.indexQuestions(currentFormVersion.descriptor);
+    const [currentQuestion, setCurrentQuestion] = useState(
+        indexedQuestions[params.questionName],
+    );
 
-        return (
-            <section className={classes.relativeContainer}>
-                <TopBar
-                    title={
-                        currentMappingVersion
-                            ? intl.formatMessage(MESSAGES.mapping, {
-                                  name: currentMappingVersion.form_version.form
-                                      .name,
-                                  id: currentMappingVersion.form_version
-                                      .version_id,
-                                  type: currentMappingVersion.mapping
-                                      .mapping_type,
-                              })
-                            : intl.formatMessage(MESSAGES.loading)
-                    }
-                    displayBackButton
-                    goBack={() => {
-                        if (prevPathname || !currentMappingVersion) {
-                            router.goBack();
-                        } else {
-                            redirectToReplace(baseUrls.mappingsPath, {});
-                        }
-                    }}
-                />
-                {fetching && <LoadingSpinner />}
+    if (
+        currentFormVersion &&
+        params.questionName &&
+        currentQuestion == undefined
+    ) {
+        setCurrentQuestion(indexedQuestions[params.questionName]);
+    }
```

Comment:
Not a big fan of a conditional setting up the state on each render, maybe a useEffect is a better solution here